packages/components/src/components/hds/form/masked-input/base.ts

```diff
@@ -28,25 +28,32 @@ export interface HdsFormMaskedInputBaseSignature {
 }
 
 export default class HdsFormMaskedInputBase extends Component<HdsFormMaskedInputBaseSignature> {
-  @tracked isContentMasked = this.args.isContentMasked ?? true;
+  @tracked _isContentMasked = true;
+  @tracked _isControlled = this.args.isContentMasked !== undefined;
+
+  get isContentMasked(): boolean {
```

Comment:
I did a quick test myself, to see if I was maybe missing something, but actually it doesn't work as intended: as you can see from this clip, if you mix toggling the state via internal and external control, things don't work fine (essentially, once you set the component in controlled mode, then the internal interaction is ignored; but as a end-user, the component seems broken):

https://github.com/user-attachments/assets/cf3eb9eb-bd8b-49c5-8d07-c569a2fea03c

Here is the patch, if you want to use it yourself to play with it, and see if you can find a workaround: 
[patch.diff.zip](https://github.com/user-attachments/files/17562003/patch.diff.zip)
2258c98-7fc2-4a1b-ba27-1375ce25d8c9
but as far as I can tell (and I remember), there is no way to use the arg to determine the initial state as well as if it's controlled or not, while at the same time preserving the ability to change the state internally. And even if we rely on the knowledge of when a user clicks the button, and we switch to an internal state, at that point the external state goes out of sync with the internal one (hence, my comment above)

That's why it's so complex this thing, and needs to be super-tested thoroughly (I spent days on this, for the pagination).

My 2 cents is to remove this logic for external state control, to avoid inadvertently releasing breaking changes to the consumers.

Reply:
[btw] please don't take everything I said above at face value, but verify it: it well may be that I'm totally wrong, or I missed something, or my test is not correct

Reply:
I did a similar test and it worked. I think @didoo's suggestion to have an example in the showcase would make sense and help us all have a common reference and instance to test with.

Reply:
> I did a similar test and it worked

🤯

Reply:
I may have missed a certain combination of interactions though

Reply:
I tend to agree with @didoo here, controlling the state from the outside seems like a feature we don't need to introduce now.

Reply:
Resolving this because we ended up removing the controlled logic from this PR.